File: src/main/java/kave/IClientAPI.java
package kave;

import java.io.File;
import java.io.IOException;
import java.util.List;

import org.openqa.selenium.remote.JsonException;

public interface IClientAPI {

	List<ModelDescriptor> getIndex() throws IOException, JsonException;

	Result<String> upload(ModelDescriptor fileDescriptor, File file, CollisionHandling collionCase) throws IOException;

	Result<File> download(ModelDescriptor md) throws IOException;
	
	boolean saveFile(ModelDescriptor md, File file) throws IOException;

	Result<String> delete(ModelDescriptor md) throws IOException;
}

File: src/main/C#/ClientAPI/ClientAPI/IClientAPI.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClientAPI
{
    interface IClientAPI
    {
        List<ModelDescriptor> getIndex();

        Result<String> upload(ModelDescriptor fileDescriptor, FileInfo file, CollisionHandling collionCase);

        Result<FileInfo> download(ModelDescriptor md);
	
	    bool saveFile(ModelDescriptor md, FileInfo file);

        Result<String> delete(ModelDescriptor md);

    }
}
